### PR TITLE
Player 590 2

### PR DESCRIPTION
**Pull request by dallen42**
Required to report discovery analytics with the JSON pipeline

### CODE REVIEW

**Comment by dallen42** [comment]
Don't merge this quite yet, but please begin review. 

Note that this basically separates sendDiscoveryDisplayEvent into two different functions, sendDiscoveryDisplayEvent and sendDiscoveryDisplayEventOld. The sendDiscoveryDisplayEventOld method is for compatibility with other code that requires the display event to include a list of all displayed assets, where the new sendDiscoveryDisplayEvent sends a separate event for each asset as required by the analytics framework. Will need to refactor this to be a bit cleaner.

File: js/components/utils.js [hunk 1]
```diff
@@ -165,6 +165,94 @@ var Utils = {
...
+        enc[j] = b64.indexOf(s.charAt(i++));
+      }
+      results += String.fromCharCode((enc[0] << 2) | (enc[1] >> 4),
+                                      enc[2] == 64 ? 0 : ((enc[1] & 15) << 4) | (enc[2] >> 2),
+                                      enc[3] == 64 ? 0 : ((enc[2] & 3) << 6) | enc[3]);
+    } while (i < s.length);
+
+    //trim tailing null characters
+    return results.replace(/\0/g, "");
+  },
```
Comment (zelibobla) [inline js/components/utils.js]: https://caniuse.com/#search=atob
https://help.ooyala.com/video-platform/concepts/pbv4_environments.html
Seems like `atob` exists in all officially supported by Ooyala browsers. So custom implementation is redundant.

**Comment by dallen42** [comment]
Refactored the code a bitand updated unit tests, it should be ok to merge now

File: js/components/utils.js (hunk 1 above)
Comment (dallen42) [reply js/components/utils.js]: @zelibobla even so, I think it is worth it to include this helper to have as broad support as possible. Customers could report issues if their user base is watching with older or unsupported browsers and this is a cheap defense against that

**Comment by zelibobla** [comment]
On my local it works with no issues:
```
$ git clone https://github.com/ooyala/html5-skin.git
Cloning into 'html5-skin'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 25786 (delta 75), reused 65 (delta 45), pack-reused 25659
Receiving objects: 100% (25786/25786), 7.27 MiB | 2.90 MiB/s, done.
Resolving deltas: 100% (18751/18751), done.

$ cd html5-skin/
$ git checkout PLAYER-590-2
Branch 'PLAYER-590-2' set up to track remote branch 'PLAYER-590-2' from 'origin'.
Switched to a new branch 'PLAYER-590-2'

$ npm install
> fsevents@1.2.4 install /Users/anton_zelenskii/Projects/killme/html5-skin/node_modules/fsevents
> node install
[fsevents] Success: "/Users/anton_zelenskii/Projects/killme/html5-skin/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile
> node-sass@4.9.4 install /Users/anton_zelenskii/Projects/killme/html5-skin/node_modules/node-sass
> node scripts/install.js
Cached binary found at /Users/anton_zelenskii/.npm/node-sass/4.9.4/darwin-x64-57_binding.node
> node-sass@4.9.4 postinstall /Users/anton_zelenskii/Projects/killme/html5-skin/node_modules/node-sass
> node scripts/build.js
Binary found at /Users/anton_zelenskii/Projects/killme/html5-skin/node_modules/node-sass/vendor/darwin-x64-57/binding.node
Testing binary
Binary is fine
added 1900 packages from 1213 contributors in 25.833s

$ git submodule init
Submodule 'config' (https://github.com/ooyala/skin-config.git) registered for path 'config'

$ git submodule update
Cloning into '/Users/anton_zelenskii/Projects/killme/html5-skin/config'...
Submodule path 'config': checked out '14ff5e4676cdf21a27b6f4763a37c114e95befa1'

$ git pull
Already up to date.
```
 Shall you have any issues further please contact me in Slack.

**Comment by dallen42** [comment]
@zelibobla if there are no other issues, can you approve and merge?

**Pull request merged**